### PR TITLE
CRDT labs

### PR DESCRIPTION
**Pull request by niwinz**

### CODE REVIEW

File: src/cats/protocols.cljc [hunk 1]
```diff
@@ -97,3 +97,9 @@
 (defprotocol MonadTrans
   "A monad transformer abstraction."
   (-lift [m mv] "Lift a value from the parameterized monad to the transformer."))
+
+;; More info: http://en.wikipedia.org/wiki/Semilattice
+
+(defprotocol JoinSemiLattice
+  "A algebraic structure with element joins."
+  (-join [_ other] "Converge two data structures."))
```
Comment (dialelo) [inline src/cats/protocols.cljc]: Note that there is `cats.core/join` which is used for removing one monadic layer, so we may want to call the join on semilattices differently.

Comment (niwinz) [inline src/cats/protocols.cljc]: Or rename join to mjoin xD, or just do not have any public api on cats.core for semilattice...

**Comment by niwinz** [comment]
I think that for first itaration this PR is ready. @dialelo @yurrriq feel free to review, comment and merge!

**Comment by dialelo** [comment]
Really cool stuff, we can make a list of other candidate CRDTs to implement in an issue too.

**Pull request merged**